corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -587,12 +587,30 @@ def save_deprecated_form(form):
 
     @staticmethod
     @transaction.atomic
-    def update_form_problem_and_state(form):
-        with get_cursor(XFormInstanceSQL) as cursor:
-            cursor.execute(
-                'SELECT update_form_problem_and_state(%s, %s, %s)',
-                [form.form_id, form.problem, form.state]
-            )
+    def update_form(form):
+        from corehq.form_processor.change_publishers import publish_form_saved
+        from corehq.sql_db.util import get_db_alias_for_partitioned_doc
+        assert form.is_saved(), "this method doesn't support creating unsaved forms"
+
+        db_name = get_db_alias_for_partitioned_doc(form.form_id)
+        if form.orig_id:
+            old_db_name = get_db_alias_for_partitioned_doc(form.orig_id)
+            assert old_db_name == db_name, "this method doesn't support moving the form to new db"
+
+        form_id_updated = form.orig_id != form.form_id and form.orig_id is not None
```

Comment:
that's not a good general test since it is only true when `orig_id` get's set. If the form get's updated after that (for whatever reason) then it won't hold true.